docs/protos/typescript.md

```diff
@@ -33,81 +36,280 @@ There are a few inputs that the SDK needs in order to be properly configured.
-import { createConnectTransport } from '@connectrpc/connect-node';
+// authenticates with Neosync Cloud
+const neosyncClient = getNeosyncClient({
+  getAccessToken: () => {
+    return 'neo_at_v1_xxxxxxxxxxxx'; // API key
+  },
+  getTransport(interceptors) {
+    return createConnectTransport({
+      baseUrl: 'https://neosync-api.svcs.neosync.dev', // base url
+      httpVersion: '2',
+      interceptors: interceptors, // interceptors
+    });
+  },
+});
+
+// sample data object
+const data = {
+  user: {
+    name: 'John Doe',
+    email: 'john@example.com',
+  },
+  details: {
+    address: '123 Main St',
+    phone: '555-1234',
+    favorites: ['dog', 'cat', 'bird'],
+  },
+};
+
+const transformers: TransformerMapping[] = [
+  new TransformerMapping({
+    expression: '.user.name', // targets the name field in the user object with a jq expression
+    transformer: new TransformerConfig({
+      config: {
+        case: 'generateFullNameConfig', // sets the generateFullNameConfig
+        value: new GenerateFullName({}), // sets the GenerateFullName transformer
+      },
+    }),
+  }),
+  new TransformerMapping({
+    expression: '.user.email', // targets the email field in the user object with a jq expression
+    transformer: new TransformerConfig({
+      config: {
+        case: 'generateEmaileConfig', // sets the generateEmailConfig
+        value: new GenerateEmail({}), // sets the GenerateEmail transformer
+      },
+    }),
+  }),
+  new TransformerMapping({
+    expression: '.details.address', // targets the address field in the details object with a jq expression
+    transformer: new TransformerConfig({
+      config: {
+        case: 'generateFullAddressConfig', // sets the generateFullAddressConfig
+        value: new GenerateFullAddress({}), // sets the GenerateFullAddress transformer
+      },
+    }),
+  }),
+  new TransformerMapping({
+    expression: '.details.phone', // targets the phone field in the details object with a jq expression
+    transformer: new TransformerConfig({
+      config: {
+        case: 'generateStringPhoneNumberConfig', // sets the generateStringPhoneNumberConfig
+        value: new GenerateStringPhoneNumber({
+          // sets the GenerateStringPhoneNumber transformer
+          max: BigInt(12), // sets the max number of digits in the string phone number
+          min: BigInt(9), // sets the min number of digits in the string phone number
+        }),
+      },
+    }),
+  }),
+];
+
+async function runAnonymization() {
+  try {
+    const result: AnonymizeSingleResponse =
+      await neosyncClient.anonymization.anonymizeSingle({
+        inputData: JSON.stringify(data), // stringify the data object from above
+        transformerMappings: transformers, // pass in your transformer mappings that you defined
+      });
+    console.log('Anonymization result:', result.outputData);
+  } catch (error) {
+    console.error('Error:', error);
+  }
+}
+
+// calling our async function
+runAnonymization();
```

Comment:
It would be helpful to update this line to actually await the promise as otherwise the script will exit immeditaely and not wait for the response.
Maybe I'm wrong though.

Something like:
```suggestion
runAnonymization().then(() => console.log("script completed").catch(console.error);
```

Reply:
i guess there could be an error that isn't caught by the internal try-catch so you're prob right that I should just handle that outside of the function - but fwiw i think the presence of the await inside of the try catch will wait for the response - will update anyways